### DISCUSSION TITLE
Prevent Cypress capturing Cross Origin errors

### DISCUSSION THREAD

**Discussion by trinode** [cypress-io/cypress#15652] [Questions and Help]
We're migrating a large site over to cypress but it seems to be sensitive to third party scripts failing for Cross origin reasons - puppeteer carries on fine.

Is there a way in cypress to get it to stop failing if a cross origin error occurs (it's not the main page navigating out, but a third party script , and those failures aren't something we really care about, they're included by some scripts we depend upon but have no control over - the page would normally function fine but with an error in the console, but cypress catches it and fails completely).

These scripts are not necessary for us during development / testing of the things we're developing - the versions we pull in are the same as in production too, if something comes along during test, they've already broken production.

We can run with web security disabled, so the browser lets the scripts run, and cypress doesn't see a cross origin error. This however ties us into chrome, negating one of the benefits of cypress.

There is a blocked script and a blocked modification to `document`;

Ideally we could be able to allow those cross origin errors to happen (in the browser) and ignore them (in cypress).

Something like `Cypress.on('cross-origin:exception', () => false);` would allow use to get going, and it'd be a conscious decision by us to acknowledge we are ignoring a class of error.

**Comment by lmiller1990** [comment]
You could do something like [suggested here](https://github.com/cypress-io/cypress/issues/2554#issuecomment-442334752).

```js
Cypress.on('uncaught:exception', (err, runnable) => {
  // returning false here prevents Cypress from
  // failing the test
  return false
})
```

Your tests will still fail due to errors raised by Cypress assertions (for example, you cannot find button, element, etc).

**Reply by trinode** [reply]
Hi, thanks for looking at it.

I expected that to work too! That was my first attempt after reading the docs (so I proposed a very similar approach using  `cross-origin:exception` as the key for `Cypress.on` to allow cross origin errors to be caught the same as uncaught exceptions at the end of my previous post). 

After it didn't work, I presumed that it was expected behaviour that the cross origin request failure wasn't handled under `uncaught:exception`, I think this kind of error is treated differently somehow?

**Reply by lmiller1990** [reply]
Odd, it's working fine for me. Will need a repro (see Jennifer's comment below) to debug more.

**Comment by jennifer-shehane** [comment]
There's a proposal to only warn on these uncaught exceptions which I would love to see implemented. https://github.com/cypress-io/cypress/issues/4933

If there's a case where the `uncaught:exception` handler is not catching the event, we'd need a full example to investigate why that would be.
